nixos/modules/services/web-apps/stirling-pdf.nix

```diff
@@ -0,0 +1,104 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.stirling-pdf;
+in
+{
+  options.services.stirling-pdf = {
+    enable = lib.mkEnableOption "the stirling-pdf service";
+
+    package = lib.mkPackageOption pkgs "stirling-pdf" { };
+
+    environment = lib.mkOption {
+      type = lib.types.attrsOf (lib.types.nullOr (lib.types.either lib.types.str lib.types.path));
+      default = { };
+      example = {
+        SERVER_PORT = "8080";
+        INSTALL_BOOK_AND_ADVANCED_HTML_OPS = "true";
+      };
+      description = ''
+        Environment variables for the stirling-pdf app.
+        See https://github.com/Stirling-Tools/Stirling-PDF#customisation for available options.
+      '';
+    };
+
+    environmentFile = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+      description = ''
+        File containing additional environment variables to pass to Stirling PDF.
```

Comment:
Maybe highlight that secrets should use `environmentFile`?

Also, what about making it `environmentFiles` (`type = listOf path`, `default = []`).